tracer/src/Datadog.Trace/Iast/Aspects/System.Text/StringBuilderAspects.cs

```diff
@@ -0,0 +1,199 @@
+        return StringBuilderModuleImpl.OnStringBuilderAppend(target.Append(value?.ToString(), startIndex, count), initialLength, value, value?.Length ?? 0, startIndex, count);
+    }
+#endif
+
+    /// <summary>  StringBuilder.Append aspect </summary>
+    /// <param name="target"> StringBuilder instance </param>
+    /// <param name="value"> string parameter </param>
+    /// <param name="startIndex"> startIndex parameter </param>
+    /// <param name="charCount"> charCount parameter </param>
+    /// <returns> instance.Append() </returns>
+    [AspectMethodReplace("System.Text.StringBuilder::Append(System.Char[],System.Int32,System.Int32)")]
+    public static StringBuilder Append(StringBuilder target, char[]? value, int startIndex, int charCount)
+    {
+        var initialLength = target.Length;
+        return StringBuilderModuleImpl.OnStringBuilderAppend(target.Append(value, startIndex, charCount), initialLength, value, value?.Length ?? 0, startIndex, charCount);
+    }
+
+    /// <summary>  StringBuilder.Append aspect </summary>
+    /// <param name="target"> StringBuilder instance </param>
+    /// <param name="value"> string parameter </param>
+    /// <returns> instance.Append() </returns>
+    [AspectMethodReplace("System.Text.StringBuilder::Append(System.Object)")]
+    public static StringBuilder Append(StringBuilder target, object? value)
+    {
+        var initialLength = target.Length;
+
+        object? valueObject;
+        int length;
+        if (value is StringBuilder valueStringBuilder)
+        {
+            valueObject = valueStringBuilder;
+            length = valueStringBuilder!.Length;
+        }
+        else
+        {
+            valueObject = value?.ToString();
+            length = (valueObject as string)?.Length ?? 0;
+        }
+
+        return StringBuilderModuleImpl.OnStringBuilderAppend(target.Append(value), initialLength, valueObject, length, 0, length);
+    }
+
+    /// <summary>  StringBuilder.Append aspect </summary>
+    /// <param name="target"> StringBuilder instance </param>
+    /// <param name="value"> string parameter </param>
+    /// <returns> instance.Append() </returns>
+    [AspectMethodReplace("System.Text.StringBuilder::Append(System.Char[])")]
+    public static StringBuilder Append(StringBuilder target, char[]? value)
+    {
+        var initialLength = target.Length;
+        var length = value?.Length ?? 0;
+        return StringBuilderModuleImpl.OnStringBuilderAppend(target.Append(value), initialLength, value, length, 0, length);
+    }
+
+    /// <summary>  StringBuilder.AppendLine aspect </summary>
+    /// <param name="target"> StringBuilder instance </param>
+    /// <param name="value"> string parameter </param>
+    /// <returns> instance.AppendLine() </returns>
+    [AspectMethodReplace("System.Text.StringBuilder::AppendLine(System.String)", AspectFilter.StringLiteral_1)]
+    public static StringBuilder AppendLine(StringBuilder target, string? value)
```

Comment:
Looking at the other available `Append()` etc methods, I wonder if we need to handle these?:

- https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append?view=net-7.0#system-text-stringbuilder-append(system-text-stringbuilder-appendinterpolatedstringhandler@)
- https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append?view=net-7.0#system-text-stringbuilder-append(system-iformatprovider-system-text-stringbuilder-appendinterpolatedstringhandler@)
- https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append?view=net-7.0#system-text-stringbuilder-append(system-readonlyspan((system-char)))
- https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append?view=net-7.0#system-text-stringbuilder-append(system-readonlymemory((system-char)))
- https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append?view=net-7.0#system-text-stringbuilder-append(system-object)

I wonder about `CopyTo()`, `Replace`, `Insert`, and `GetChunks()` too, as they seem like ways for a tainted string to get in there?

Reply:
Right now, we only taint strings, some char[] cases and stringBuilder objects, sow e do not cover the overloads that include other object types. We do cover Append(object) because a string o stringBuilder can be sent as an object.

On the other side, those other methods (Insert, replace, etc.) will not be covered for now. It has been decided by all the IAST teams to cover only some methods and leave others for later. Some, like AppendFormat, are so complex that we will probably taint the whole result if some of the inputs is tainted instead of analyzing the tainting and calculating the ranges, but it has not been decided yet.